corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -842,32 +832,59 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", [
             urlObject.setRequestInitiatedByTag(initiatedByTag);
             var fetchingPrompts = FormplayerFrontend.getChannel().request("app:select:menus", urlObject);
             $.when(fetchingPrompts).done(function (response) {
-                // Update models based on response
-                if (response.queryResponse) {
-                    _.each(response.queryResponse.displays, function (responseModel, i) {
-                        self._getChildModels()[i].set({
-                            error: responseModel.error,
-                            required: responseModel.required,
-                            required_msg: responseModel.required_msg,
-                        });
-                    });
-                } else {
-                    _.each(response.models, function (responseModel, i) {
-                        const childModels = self._getChildModels();
-                        childModels[i].set({
-                            error: responseModel.get('error'),
-                            required: responseModel.get('required'),
-                            required_msg: responseModel.get('required_msg'),
-                        });
-                    });
-                }
+                self.updateModels(response);
                 promise.resolve(response);
 
             });
             sessionStorage.validationInProgress = false;
             return promise;
         },
 
+        displayErrors: function () {
+            var self = this;
+            // Gather error messages
+            var invalidFields = [];
+            self._getChildren().forEach(function (childView) {
```

Comment:
@orangejenny I am running across an issue where `self._getChildren()` returns me an empty [] when I run search due to which error display doesn't work correctly. On looking at the code this method populates children using `this.children` which returns empty to me. Any idea how can I debug this ? I am quite blocked on figuring out when and where `this.children` gets its value from.

Reply:
You probably know `this.children` is populated by Marionette, it's part of this being a `CollectionView`. The view lifecycle is event driven, so it can be tricky to debug, but there are callbacks for the various events. Check out the [CollectionView docs](https://marionettejs.com/docs/master/marionette.collectionview.html) and the more general [View lifecycle docs](https://marionettejs.com/docs/master/view.lifecycle.html). (Make sure you're in the right version of docs - a search engine took me to the v2 docs, but we're on v4 and this area has changed a good deal.)

I poked around on this branch, just adding console statements in `executeSearch`, `performSubmit`, etc. and adding a few handlers to the `QueryListView` like `onBeforeDestroy` that just contained console statements. Seems the `QueryListView` is getting destroyed as part of the `menu:query` response (because it calls `listMenus` which eventually calls `showMenu` and then [this code](https://github.com/dimagi/commcare-hq/blob/86765a9002d635e952159dae5e7d40e57c6a8b82/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js#L134-L141) calls `show` on the main and sidebar regions, and `show` causes the view to get destroyed and a new one created). I believe that's why `children` is empty when this code runs. But I'm not immediately sure how to fix it. You might be able to do something like moving `displayErrors` into `onRender` or `onRenderChildren`.

Reply:
Thanks, that gives me a bunch of context to proceed on this.

Reply:
I like the idea of having some sort of explicit signal indicating whether there has been a user-initiated submission.  Can we get that information from the formplayer response?  Is that what you're suggesting with the response type idea?